numpy/ma/core.py

```diff
@@ -901,22 +901,32 @@ class _MaskedBinaryOperation:
         Filling value for the first argument, default is 0.
     filly : scalar, optional
         Filling value for the second argument, default is 0.
-
+    op    : bool, optional
+        Set to true when used in (non-reversed) binary operators.
     """
-    def __init__ (self, mbfunc, fillx=0, filly=0):
+    def __init__ (self, mbfunc, fillx=0, filly=0, op=False):
         """abfunc(fillx, filly) must be defined.
            abfunc(x, filly) = x for all x to enable reduce.
         """
         self.f = mbfunc
         self.fillx = fillx
         self.filly = filly
+        self.op = op
         self.__doc__ = getattr(mbfunc, "__doc__", str(mbfunc))
         self.__name__ = getattr(mbfunc, "__name__", str(mbfunc))
         ufunc_domain[mbfunc] = None
         ufunc_fills[mbfunc] = (fillx, filly)
 
+    def __get__(self, obj, type=None):
```

Comment:
I'm slightly confused why this is being done. It makes this a descriptor, right?